ADMVEDU/ViewControllers/FilterViewController.swift

```diff
@@ -0,0 +1,242 @@
+        textField.inputView = pickerView
+        textField.textAlignment = .center
+        pickerView.delegate = self
+        pickerView.dataSource = self
+        pickerView.tag = tag
+        stackView.addArrangedSubview(textField)
+    }
+
+    func getCountryName(countryCode: String) -> String? {
+        let current = Locale(identifier: "en_us")
+        return current.localizedString(forRegionCode: countryCode)
+    }
+
+    private func getUniqueIdentifier() -> Int {
+        uniqueIdentifier += 1
+        return uniqueIdentifier
+    }
+
+    @objc
+    func switchTriggered(sender: AnyObject) {
+        if requestOptions.explisit == .yes {
+            requestOptions.explisit = .no
+        } else {
+            requestOptions.explisit = .yes
+        }
+    }
+}
+
+extension FilterViewController: UITextFieldDelegate {
+}
+
+extension FilterViewController: UIPickerViewDelegate, UIPickerViewDataSource {
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        switch pickerView.tag {
+        case countryPicker.tag:
+            return Country.allCases.count
+        case mediaPicker.tag:
+            return MediaType.allCases.count
+        default:
+            return 0
+        }
+    }
+
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        switch pickerView.tag {
+        case countryPicker.tag:
+            let items = Country.allCases.map { $0.rawValue }
+            return getCountryName(countryCode: items[row])
+        case mediaPicker.tag:
+            let items = MediaType.allCases.map { $0.rawValue }
+            return items[row]
+        default:
+            return "Not Found"
+        }
+    }
+
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        switch pickerView.tag {
+        case countryPicker.tag:
+            let items = Country.allCases.map { $0.rawValue }
+            countryTextField.text = getCountryName(countryCode: items[row])
+            countryTextField.resignFirstResponder()
+            requestOptions.country = items[row]
+        case mediaPicker.tag:
+            let items = MediaType.allCases.map { $0.rawValue }
+            mediaTextField.text = items[row]
+            mediaTextField.resignFirstResponder()
+            requestOptions.media = MediaType.allCases[row]
+
+        default:
+            return
+        }
+    }
+}
+
+extension FilterViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = UITableViewCell(style: UITableViewCell.CellStyle.default, reuseIdentifier: "cell")
```

Comment:
why not dequeueReusableCell?

Reply:
I use a TableView to display only one cell with a Slider. Is it better to replace it with dequeueReusableCell ?

Reply:
if you want to reuse a slider, it should be a custom view
table view is not for a single elements (though technically it's possible but there are better approaches)